chiron/mcmc.py

```diff
@@ -112,103 +175,802 @@ def run(
+        self.displacement_sigma = displacement_sigma
+
+        self.atom_subset = atom_subset
+        self.atom_subset_mask = None
+
+    def _report(
+        self,
+        step: int,
+        iteration: int,
+        elapsed_step: int,
+        acceptance_probability: float,
+        sampler_state: SamplerState,
+        thermodynamic_state: ThermodynamicState,
+        nbr_list: Optional[PairsBase] = None,
+    ):
+        """
+        Report the current state of the MC displacement move.
+
+        Parameters
+        ----------
+        step : int
+            The current step of the simulation move.
+        iteration : int
+            The current iteration of the move sequence (i.e., how many times has this been called thus far).
+        elapsed_step : int
+            The total number of steps that have been taken in the simulation move. step+ nr_moves*iteration
+        acceptance_probability : float
+            The acceptance probability of the move.
+        sampler_state : SamplerState
+            The sampler state of the system.
+        thermodynamic_state : ThermodynamicState
+            The thermodynamic state of the system.
+        nbr_list : Optional[PairBase]=None
+            The neighbor list or pair list for evaluating interactions in the system, default None
+
+        """
+        potential = thermodynamic_state.potential.compute_energy(
+            sampler_state.x0, nbr_list
+        )
+        self.reporter.report(
+            {
+                "step": step,
+                "iteration": iteration,
+                "elapsed_step": elapsed_step,
+                "potential_energy": potential,
+                "displacement_sigma": self.displacement_sigma.value_in_unit_system(
+                    unit.md_unit_system
+                ),
+                "acceptance_probability": acceptance_probability,
+            }
+        )
+
+    def _update_stepsize(self):
+        """
+        Update the displacement_sigma to reach a target acceptance probability of 0.5.
+        """
+        acceptance_ratio = self.n_accepted / self.n_proposed
+        if acceptance_ratio > 0.6:
+            self.displacement_sigma *= 1.1
+        elif acceptance_ratio < 0.4:
+            self.displacement_sigma /= 1.1
+
+    def _propose(
+        self,
+        current_sampler_state: SamplerState,
+        current_thermodynamic_state: ThermodynamicState,
+        current_reduced_pot: float,
+        current_nbr_list: Optional[PairsBase] = None,
+    ) -> Tuple[SamplerState, ThermodynamicState, float, float, Optional[PairsBase]]:
+        """
+        Implement the logic specific to displacement changes.
+
+        Parameters
+        ----------
+        current_sampler_state : SamplerState, required
+            Current sampler state.
+        current_thermodynamic_state : ThermodynamicState, required
+            Current thermodynamic state.
+        current_reduced_pot : float, required
+            Current reduced potential.
```

Comment:
It might be easier if we are explicit here and add energy.

Reply:
See my prior comment. I don't think we should add energy to the name.